backend/tests/exploregens.spec.ts

```diff
@@ -324,16 +329,27 @@ describe('exploregens unit test', () => {
         });
 
         it("location is a valid string", () => {
-            workspaceConfigMock.expects("get").withExactArgs(ExploreGens["INSTALLATION_LOCATION"]).returns(TESTVALUE);
+            const customLocation = TESTVALUE;
+            workspaceConfigMock.expects("get").withExactArgs(ExploreGens["INSTALLATION_LOCATION"]).returns(customLocation);
+            fsMock.expects("existsSync").withExactArgs(customLocation).returns(true);
             const res = exploregens["getGeneratorsLocationParams"]();
             expect(res).to.be.equal(`--prefix ${TESTVALUE}`);
         });
 
         it("location is a string with unnecessary spaces", () => {
-            workspaceConfigMock.expects("get").withExactArgs(ExploreGens["INSTALLATION_LOCATION"]).returns(`   ${TESTVALUE}   `);
+            const customLocation = `   ${TESTVALUE}   `;
+            workspaceConfigMock.expects("get").withExactArgs(ExploreGens["INSTALLATION_LOCATION"]).returns(customLocation);
+            fsMock.expects("existsSync").withExactArgs(TESTVALUE).returns(true);
             const res = exploregens["getGeneratorsLocationParams"]();
             expect(res).to.be.deep.equal(`--prefix ${TESTVALUE}`);
         });
+
+        it("location starts with tild ~", () => {
```

Comment:
this test should have fsMock.expects("existsSync").withExactArgs(TESTVALUE).returns(false);

Reply:
**Better** not mock it. location is invalid so existSync will return false.

Reply:
In test environment accidentally this path may exist and the test will fail. It is better to control it with mock

Reply:
0.01% that accidentally contains a path equal to '~/notExistLocation'

Reply:
It is always better 100%, especially for tests.

Reply:
so, in this case it 100% that there '~/notExistLocation' will not exist on test system.